mmdet/apis/ote/extension/datasets/mmdataset.py

```diff
@@ -194,3 +198,25 @@ def get_ann_info(self, idx):
             # For RepeatDataset wrapper.
             label_list = self.dataset.CLASSES
         return get_annotation_mmdet_format(dataset_item, label_list)
+
+    def filter_small_gt(self, item: dict) -> dict:
+        """
+        Function to filter instances in DatasetItem if its width or height is smaller than self.min_size
+
+        :param item: 'data_info' dict that represents the DatasetItem
+        :return dict: the same dict with filtered instances
+        """
+        dataset_item = item['dataset_item']
+        width, height = dataset_item.width, dataset_item.height
+        filtered_anns = []
+
+        for ann in dataset_item.get_annotations():
+            box = ann.shape
+            if min(box.width * width, box.height * height) >= self.min_size:
+                filtered_anns.append(ann)
+        if len(filtered_anns) == 0:
+            print(f'All instances on the image are smaller than min_size={self.min_size} - the image was skipped')
+
+        dataset_item.annotation_scene.annotations = filtered_anns
```

Comment:
Is it necessary to change dataset item here? And `dataset_item.annotation_scene` should not be used directly in a normal situation.

Reply:
I thought about creating new dataset_item here based on filtered annotations, but it would require a few more imports. If `dataset_item.annotation_scene` should not be used directly, I'll create new dataset_item here

Reply:
Done